components/infobox/commons/infobox_team.lua

```diff
@@ -178,17 +178,17 @@ function Team:createInfobox()
 	infobox:bottom(self:createBottomContent())
 
 	if self:shouldStore(args) then
-		infobox:categories('Teams')
-		infobox:categories(unpack(self:getWikiCategories(args)))
+		infobox:addCategories('Teams')
+		infobox:addCategories(unpack(self:getWikiCategories(args)))
 	end
 
-	local builtInfobox = infobox:widgetInjector(self:createWidgetInjector()):build(widgets)
-
 	if self:shouldStore(args) then
 		self:_setLpdbData(args, links)
 		self:defineCustomPageVariables(args)
 	end
 
+	local builtInfobox = infobox:widgetInjector(self:createWidgetInjector()):build(widgets)
```

Comment:
Changing the order of building and storing will trigger the error message https://github.com/Liquipedia/Lua-Modules/blob/1dfd4bb28c988a5be5b4abafc2e405fc05340df9/components/infobox/commons/infobox_team.lua#L288-L291
in case a wiki overrides the earningcell name, e.g. on CS or AoE.

Reply:
ugh, I hate how everything has an assumed execution order

Reply:
Oh, this actually breaks on any wiki, not just on these overriding stuff

Reply:
Yeah, refactoring the earnings in the Team Infobox. Will push it into another PR for ease of review.

Reply:
#2361 
Likely gonna need one for Player/Person as well. And will go through all customs to check which ones are having too much logic in the display code.